### PR TITLE
`Playground`: Side-by-Side Expert Evaluation

### PR DESCRIPTION
**Pull request by DominikRemo**
### Motivation and Context
By GPT-4o and Suno (sound on 🎶)

https://github.com/user-attachments/assets/f30f168f-db78-4baf-abae-dfbdccca7076

The primary motivation behind this tool is to create a robust and versatile benchmark for evaluating feedback on student submissions across specific use cases.  

- **Sustainability:** The tool is designed to be flexible and adaptable, allowing benchmarks to be automatically repeated with newer or alternative large language models (LLMs) or other feedback generation methods as they become available.
- **Reliability:** By ensuring that benchmarks are tested on real-world data not included in LLM training data, the tool helps prevent overfitting and provides a more reliable evaluation of feedback quality.
- **Readability:** The benchmarks produced by this tool are designed to be clear and understandable, with metrics that are specific to the use case, making it easier for researchers and practitioners to interpret and apply the results effectively

### Description

This PR introduces the Side-by-Side Evaluation Tool, designed to assist experts in evaluating feedback provided on student submissions. This tool is especially useful for researchers seeking to assess the quality and relevance of feedback across multiple criteria and through multiple evaluators. Ths Side-by-Side Tool provides two views:

- **Researcher View:** The researcher defines evaluation metrics (e.g., accuracy, tone), selects exercises and submissions for review, and generates individual links for each expert involved in the evaluation.
- **Expert View:** Each expert reviews submissions and evaluates feedback based on the metrics set by the researcher. A Likert scale is used for each metric, allowing for consistent and structured assessments across feedback instances.

### Steps for Testing

1. Access the playground on the test server or locally
2. Select **Evaluation Mode**, and scroll down to the **Expert Evaluation**
3. Create a new evaluation by providing an Evaluation Name
4.  Import exercises using the jsons below :arrow_down: 
5. Add at least two Metrics (e.g. Accuracy, Adaptability, Tone...) by providing a Title, Summary and a Description
6. Create a new expert link and **Save Changes**
7. **Start the Evaluation** and open the expert link
8. See the **Welcome Screen** and proceed to the **Tutorial**
9. Make sure the tutorial is understandable and **Start Evaluation**
10. Evaluate the submissions according to how it is described in the tutorial
11. Also make sure all buttons are working as expected => try **Continue Later**, exiting the page and continuing
12. Finish the evaluation by evaluating all metrics for all submissions

[](https://athena-test1.ase.cit.tum.de)
[](https://athena-test2.ase.cit.tum.de)

### Test Data
Use the following exercises to test
[exercise-5000.json](https://github.com/user-attachments/files/17716450/exercise-5000.json)
[exercise-800.json](https://github.com/user-attachments/files/17716457/exercise-800.json)
[exercise-700.json](https://github.com/user-attachments/files/17716458/exercise-700.json)
[exercise-123.json](https://github.com/user-attachments/files/17716459/exercise-123.json)

### Screenshots

**Researcher View** - Imported Exercises and Metrics
image

**Researcher View** - Defining Metrics and creating expert links
image

**Expert View** - Welcome Screen
image

**Exper View** - Evaluation View
image

### CODE REVIEW

File: playground/package.json [hunk 1]
```diff
@@ -14,6 +14,9 @@
   },
   "dependencies": {
     "@blueprintjs/core": "5.5.1",
+    "@fortawesome/fontawesome-svg-core": "^6.6.0",
+    "@fortawesome/free-solid-svg-icons": "^6.6.0",
+    "@fortawesome/react-fontawesome": "^0.2.2",
```
Comment (FelixTJDietrich) [inline playground/package.json]: Please pin the dependencies

File: playground/src/model/data_mode.ts [hunk 2]
```diff
@@ -1 +1 @@
-export type DataMode = "example" | "evaluation" | string;
+export type DataMode = "example" | "evaluation" | "expert_evaluation" | "expert_evaluation/exercises" | string;
```
Comment (FelixTJDietrich) [inline playground/src/model/data_mode.ts]: Why do you have two different modes here insted of just one mode?

**Review by FelixTJDietrich — CHANGES_REQUESTED** [review]

**Review by FelixTJDietrich — DISMISSED** [review]
Just implement the comments from above, this PR looks good to me. I also tested again on the test server and could not see any issues with it. Good job!

**Review by maximiliansoelch — DISMISSED** [review]
Please incorporate the changes requested by @FelixTJDietrich.
Otherwise, it looks good! Great job 👍

File: playground/package.json (hunk 1 above)
Comment (DominikRemo) [reply playground/package.json]: We now pinned the dependencies 👍

File: playground/src/model/data_mode.ts (hunk 2 above)
Comment (DominikRemo) [reply playground/src/model/data_mode.ts]: Good catch! This was part of a previous iteration for storing evaluations. It is not needed anymore. We have now removed it 👍

**Review by FelixTJDietrich — APPROVED** [review]
